rse-bash.Rmd

```diff
@@ -0,0 +1,3902 @@
+`mkdir` means "make directory".
+Since `thesis` is a relative path
+(i.e., does not have a leading slash, like `/what/ever/thesis`),
+the new directory is created in the current working directory:
+
+```shell
+$ ls -F
+```
+
+```text
+creatures/  data/  molecules/  north-pacific-gyre/  notes.txt  pizza.cfg  solar.pdf  thesis/  writing/
+```
+
+> **Two ways of doing the same thing**
+> Using the shell to create a directory is no different than using a file explorer.
+> If you open the current directory using your operating system's graphical file explorer,
+> the `thesis` directory will appear there too.
+> While the shell and the file explorer are two different ways of interacting with the files,
+> the files and directories themselves are the same.
+
+
+> **Good names for files and directories**
+>
+> Complicated names of files and directories can make your life painful
+> when working on the command line. Here we provide a few useful
+> tips for the names of your files.
+>
+> 1. Don't use spaces.
+>
+>    Spaces can make a name more meaningful,
+>    but since spaces are used to separate arguments on the command line
+>    it is better to avoid them in names of files and directories.
+>    You can use `-` or `_` instead (e.g. `north-pacific-gyre/` rather than `north pacific gyre/`).
+>
+> 2. Don't begin the name with `-` (dash).
+>
+>    Commands treat names starting with `-` as options.
+>
+> 3. Stick with letters, numbers, `.` (period or 'full stop'), `-` (dash) and `_` (underscore).
+>
+>    Many other characters have special meanings on the command line.
+>    We will learn about some of these during this lesson.
+>    There are special characters that can cause your command to not work as
+>    expected and can even result in data loss.
+>
+> If you need to refer to names of files or directories that have spaces
+> or other special characters, you should surround the name in quotes (`""`).
+
+
+Since we've just created the `thesis` directory, there's nothing in it yet:
+
+```shell
+$ ls -F thesis
+```
+
+### Create a text file
+
+Let's change our working directory to `thesis` using `cd`,
+then run a text editor called Nano to create a file called `draft.txt`:
+
+```shell
+$ cd thesis
+$ nano draft.txt
+```
+
+> **Which Editor?**
+>
+> When we say, "`nano` is a text editor," we really do mean "text": it can
+> only work with plain character data, not tables, images, or any other
+> human-friendly media. We use it in examples because it is one of the 
+> least complex text editors. However, because of this trait, it may 
+> not be powerful enough or flexible enough for the work you need to do
+> after this workshop. On Unix systems (such as Linux and Mac OS X),
+> many programmers use [Emacs](http://www.gnu.org/software/emacs/) or
+> [Vim](http://www.vim.org/) (both of which require more time to learn), 
+> or a graphical editor such as
+> [Gedit](http://projects.gnome.org/gedit/). On Windows, you may wish to
```

Comment:
do people still use Gedit?

Reply:
Good point. They do on Linux machines, but haven't seen it elsewhere. I'll substitute in Sublime Text, which is the only editor in the Software Carpentry instructions listed for all three of Windows, Mac and Linux.